src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -171,6 +191,9 @@ public void Dispose()
                 {
                     SafeWinHttpHandle.DisposeAndClearHandle(ref _sessionHandle);
                 }
+
+                _registeredWaitHandle?.Unregister(null);
+                _internetSettingsRegistry?.Dispose();
```

Comment:
I suggest swapping the order of these two calls. While unlikely to actually matter, it's theoretically possible that, with the current ordering, we could end up closing the safe handle that the native code is still registered to set and we could end up with a use-after-free issue.

Reply:
updated. Since we are passing `SafeHandle` to the invoke, should that grab extra reference? (e.g. prevent the Release)